report-portfolio/portfolio.Rmd

```diff
@@ -0,0 +1,234 @@
+  geom_vline(xintercept = as.numeric(as.Date("2016-11-30")), color = "blue") +
+  geom_hline(yintercept = MAR, color = "purple", linetype = "dotted") +
+  annotate(geom="text", x=as.Date("2016-11-30"), 
+           y = -.05, label = "Trump", fontface = "plain", 
+           angle = 90, alpha = .5, vjust =  1.5) +
+  ylab("percent monthly returns")
+
+portfolio_scatter
+
+```
+
+
+### Histogram
+
+```{r}
+
+  sortino_byhand %>% 
+    ggplot(aes(x = returns)) +
+    geom_histogram(alpha = 0.25, binwidth = .01, fill = "cornflowerblue") +
+    geom_vline(xintercept = MAR, color = "green") +
+    annotate(geom = "text", x = MAR, 
+             y = 10, label = "MAR", fontface = "plain", 
+             angle = 90, alpha = .5, vjust =  1)
+
+
+```
+
+### Density
+
+```{r}
+  
+  sortino_density_plot <- sortino_byhand %>% 
+    ggplot(aes(x = returns)) +
+    stat_density(geom = "line", size = 1, color = "cornflowerblue") 
+  
+  shaded_area_data <- ggplot_build(sortino_density_plot)$data[[1]] %>% 
+    filter(x < MAR)
+
+  sortino_density_plot <- 
+    sortino_density_plot + 
+  geom_area(data = shaded_area_data, aes(x = x, y = y), fill="pink", alpha = 0.5) +
+  geom_segment(data = shaded_area_data, aes(x = MAR, y = 0, xend = MAR, yend = y), 
+               color = "red", linetype = "dotted") +
+  annotate(geom = "text", x = MAR, y = 5, label = paste("MAR =", MAR, sep = ""), 
+           fontface = "plain", angle = 90, alpha = .8, vjust =  -1) +
+  annotate(geom = "text", x = (MAR - .02), y = .1, label = "Downside", 
+           fontface = "plain", alpha = .8, vjust =  -1)
+           
+sortino_density_plot
+```
+
+```{r, echo = FALSE}
+portfolio_file <- 
+  paste(params$portfolio, Sys.Date(), ".xlsx", sep = "")
+
+write.xlsx(portfolio_selected, file = portfolio_file)
+
+rmarkdown::output_metadata$set(rsc_output_files = list(portfolio_file))
+```
+
+```{r, echo = FALSE}
+subject <- paste0(params$portfolioName, " portfolio Sortino report", sep = " ")
+
+rmarkdown::output_metadata$set("rsc_email_subject" = subject)
+  
+  # embed charts and data table
+  
+sign_formatter <- formatter("span", 
+  style = x ~ style(color = ifelse(x > 0, "green", 
+    ifelse(x < 0, "red", "black"))))
+
+
+tbl <- 
+  portfolio_selected %>%
+  arrange(desc(date)) %>% 
+  head() %>% 
+  format_table(x = .,
+               list(returns = sign_formatter))
+  
+  
+  msg <- compose_email(
+    body = "
+    Hello Team,
+    
+    Here are some charts.
+  
+    Here is a scatter plot as of {Sys.Date()}: 
+
+    {add_ggplot(portfolio_scatter, width = 6, height = 6)}
+
+
+    Here is the density of returns as of {Sys.Date()}: 
+
+    {add_ggplot(sortino_density_plot, width = 6, height = 6)}
+
+    Here are the raw numbers and a spreadsheet is attached:
+
+    {tbl}
+    
+
+    Let me know if you have any questions.
+    
+
+    Thanks,
+    JKR"
+  )
+
+  rmarkdown::output_metadata$set(rsc_email_body_html = msg$html_str)
```

Comment:
I know you inherited this code, but it can be cleaned up like:

```
rmarkdown::output_metadata$set(
    rsc_email_body_html = msg$html_str,
    rsc_email_images = msg$images,
    rsc_email_attachments = list(portfolio_file)
)
```

It'd also be nice to modify the subject line

Reply:
We did change the subject line above, but I have put it in this block now for consistency